python/dgl/backend/pytorch/tensor.py

```diff
@@ -70,14 +70,24 @@ def context(input):
     return input.device
 
 def device_type(ctx):
-    return ctx.type
+    return th.device(ctx).type
```

Comment:
Why we do that?
ctx should already be th.device?

Reply:
Nop. It could be a string like `"cuda:0"`.